### PR TITLE
chore(docs): update example and demo copy buttons

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Updates to the copy buttons used in the demo and example code blocks:
- consolidate copy related buttons in to `CopyButton`
- update styling to fix issues called out in https://github.com/aws-amplify/amplify-ui/pull/1905#issuecomment-1130258047

**Screenshots**
Extra space below horizontal scroll bar. overlap of copy button on horizontal scroll bar:
Before:
image

After:
image

Example text visible through copy button:

image

image

Copy button spacing when vertical scroll bar visible:

Before:
image

After:
image

#### Issue #, if available
https://github.com/aws-amplify/amplify-ui/pull/1905#issuecomment-1130258047

#### Description of how you validated changes
Manual testing

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by 0618** [comment]
if it's only docs change, we don't add changelog

File: .changeset/rare-horses-check.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"docs": patch
+---
+
+chore(docs): update example and demo copy buttons
```
Comment (wlee221) [inline .changeset/rare-horses-check.md:5]: This changeset should be removed because `docs` isn't a public NPM package 🙏

**Comment by calebpollman** [comment]
> if it's only docs change, we don't add changelog

👍 Can we configure the `changeset-bot` to not run on _docs_ updates?

**Comment by wlee221** [comment]
> 👍 Can we configure the changeset-bot to not run on docs updates?

Not yet, I opened a feature request on their repo: https://github.com/changesets/bot/issues/44

**Review by wlee221 — APPROVED** [review]
🛥️

**Review by joebuono — APPROVED** [review]
Thanks for picking this task up!

**Pull request merged**